app/views/secrets.html

```diff
@@ -14,28 +14,39 @@
           </span>
         </h1>
       </div>
+      <div ng-if="(secrets | hashSize) > 0 || filterWithZeroResults" class="data-toolbar">
+        <div class="data-toolbar-filter">
+          <project-filter></project-filter>
+        </div>
+      </div>
     </div>
   </div><!-- /middle-header-->
   <div class="middle-content">
     <div class="container-fluid">
-      <div ng-if="!secretsLoaded" class="mar-top-xl">Loading...</div>
-      <div ng-if="secretsLoaded" class="row">
+      <div class="row">
         <div class="col-md-12">
-          <div ng-if="!secrets.length">
-            <div class="empty-state-message text-center">
-              <h2>No secrets.</h2>
-              <p>
-                No secrets have been added to project {{projectName}}.
-              </p>
-              <p ng-if="project && ('secrets' | canI : 'create') && !secrets.length">
-                <a ng-href="project/{{project.metadata.name}}/create-secret" class="btn btn-primary">Create Secret</a>
-              </p>
+          <div ng-if="(secrets | hashSize) == 0">
+            <p ng-if="!secretsLoaded">
+              Loading...
+            </p>
+            <div ng-if="secretsLoaded" class="empty-state-message text-center">
+              <div ng-if="!filterWithZeroResults">
+                <h2>No secrets.</h2>
+                <p>No secrets have been added to project {{projectName}}.</p>
+                <p ng-if="project && ('secrets' | canI : 'create') && !filterWithZeroResults">
```

Comment:
You can remove `!filterWithZeroResults` from this `ng-if` since it's already checked in the `div` above.